### PR TITLE
feature(frameworks): bentoml.onnx runner accept kwargs

### PR DESCRIPTION
**Pull request by larme**
## What does this PR address?

Sometimes it's more nature to call onnx model with keyword arguments. For example, bert's tokenizer will output a dictionary, which the original model will call with `model(**input)`. Our runner should simulate this behavior

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
cc @larme, I ran the test locally, it passed, but it still didn't run on CI. can you also fix it as well?

File: src/bentoml/_internal/frameworks/onnx.py [hunk 1]
```diff
@@ -419,17 +427,31 @@ def _process_output(outs):
...
-
-            input_names: dict[str, ONNXArgCastedType] = {
-                i.name: val for i, val in zip(self.model.get_inputs(), casted_args)
-            }
-            output_names: list[str] = [o.name for o in self.model.get_outputs()]
-            raw_outs = self.predict_fns[method_name](output_names, input_names)
-            return _process_output(raw_outs)
+        if options.use_kwargs_inputs:
+
+            def _run(self: ONNXRunnable, **kwargs: ONNXArgType) -> t.Any:
```
Comment (bojiang) [inline src/bentoml/_internal/frameworks/onnx.py:432]: Does this mean we can not call `onnx_runner.run(arg1, arg2, kwarg1=xxx)` ?

Comment (aarnphm) [reply src/bentoml/_internal/frameworks/onnx.py:432]: no I don't think so.

Comment (sauyon) [reply src/bentoml/_internal/frameworks/onnx.py:432]: Yes, it does.

Comment (larme) [reply src/bentoml/_internal/frameworks/onnx.py:432]: For every onnx model we have a fixed list of inputs with their own names. Calling by args means refer each argument by its position while calling by kwargs means refer each argument by its name. Now if we mix these 2 methods, I feel that it's very easy for user to miss some arguments.

Calling by kwargs is useful when dealing with NLP models where the tokenizer output a dictionary and the dictionary is passed to the model as kwargs. In this use case there's no additional positional argument to be passed

**Comment by larme** [comment]
@aarnphm Now the tests run on CI. The problem is that the new bert tests will missing the pytorch dependencies (I thought installing transformers will install torch by default, I was wrong).

**Review by aarnphm — DISMISSED** [review]

File: src/bentoml/_internal/frameworks/onnx.py (hunk 1 above)
Comment (aarnphm) [reply src/bentoml/_internal/frameworks/onnx.py:432]: Do we need more discussion about this?

Comment (larme) [reply src/bentoml/_internal/frameworks/onnx.py:432]: let's have a discussion on engineer meeting

**Comment by sauyon** [comment]
@larme @aarnphm what's the status of this one?

**Comment by aarnphm** [comment]
This has been on this list for a while now @larme we should circle back and get this one merge

**Pull request closed without merging**